src/Tribe/Attendee_Repository.php

```diff
@@ -1158,4 +1094,71 @@ public function trigger_update_actions( $attendee_data ) {
 			do_action( "tribe_tickets_attendee_repository_update_attendee_after_update_{$this->key_name}", $attendee_data, $this );
 		}
 	}
+
+	/**
+	 * Create an order for an attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+	 *
+	 * @return int|string|false The order ID or false if not created.
+	 */
+	public function create_order_for_attendee( $attendee_data, $ticket ) {
+		// Bail if we already have an attendee or order.
+		if ( ! empty( $attendee_data['attendee_id'] ) || ! empty( $attendee_data['order_id'] ) ) {
+			return false;
+		}
+
+		// Attempt to generate a new order.
+		$orders = tribe_tickets_orders( $this->key_name );
+
+		// Bail if provider-specific order repository not found.
+		if ( empty( $orders->key_name ) ) {
+			return false;
+		}
+
+		$order_args = [
+			'full_name'    => Arr::get( $attendee_data, 'full_name' ),
+			'email'        => Arr::get( $attendee_data, 'email' ),
+			'user_id'      => Arr::get( $attendee_data, 'user_id' ),
+			'order_status' => Arr::get( $attendee_data, 'attendee_status' ),
+			'tickets'      => [
+				[
+					'id'       => $ticket->ID,
+					'quantity' => 1,
+				],
+			],
+		];
+
+		/**
+		 * Allow filtering the order data being used to create an order for the attendee.
+		 *
+		 * @since TBD
+		 *
+		 * @param array                         $order_args    List of order data to be saved.
+		 * @param array                         $attendee_data List of additional attendee data.
+		 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+		 */
+		$order_args = apply_filters( 'tribe_tickets_attendee_repository_create_order_for_attendee_order_args', $order_args, $attendee_data, $ticket );
```

Comment:
Should we start prefixing without `tribe_` already?

Reply:
Not just yet. We need to make a decision on this as a group. Lets keep as `tribe_` for now.